### ISSUE TITLE
Empty State

### ISSUE THREAD

**Issue by SOjaHPE** [grommet/hpe-design-system#3221]
## Tasks

- [x] review notes below
- [x] collate screens referenced
- [x] research the topic on the internet [1 day max]
- [x] categorize the screens and scenarios
- [x] create problem statements for each scenario
- [x] review with team
- [x] determine which scenarios are the most important to start with
- [x] create follow on issues as needed

## Acceptance Criteria

- [x] categorized and prioritized scenarios
- [x] follow on issues created as needed

## Initial Notes

Need to provide guidance for empty states.

- [ ] guidance and code examples added to the design system site
- [x] figma support and components to allow designers to easily and consistently add empty states to their designs
- [ ] update grommet story book to support engineering
- [ ] look into this opportunity to add some artwork or graphics w/ animation or movement to make the experience more dynamic. How this could be tied to brand and our visual language. Transitions.. etc.
 
There needs to be some guidance around the common types of empty state use cases:
Examples to begin with.. 

- **day zero** new to the system
- **action empty states**  users have not yet created objects/entities. These will promote appropriate actions or enable users to use the system without forcing them to find the action buried in an action menu or toolbar.
- **manage user expectations** the system is managing the expectations of the user, eg. what you've created isn't ready yet, please come back later. "It is only day 9 of the month. Your monthly report will be ready on the 31st."
- **all tasks have been completed** the user has completed all the todo's, there are no more notifications or errors
- **conditional situations** your firmware is outdated, please update in order to provide access to...
- **no data found** your query is coming up empty, show tips to fix this eg. please adjust your keywords, in other situations the user needs to clear some applied filters to see results, the system should aid the user in these situations

The guidance should cover details around layout, icon usage, CTA button, color usage, and voice and tone have not been defined for the empty state, leading to inconsistencies across the platform and other services.  It should also cover details around different data collections, do we preserve the toolbar after the last record has been deleted and render an empty state message in the body? 

This is a common pattern that is low risk, we just need to come up with a direction and provide guidance to get this consistently implemented.

**Comment by SOjaHPE** [comment]
Reference -

- https://carbondesignsystem.com/patterns/empty-states-pattern/
- https://m1.material.io/patterns/empty-states.html#
- https://www.nngroup.com/articles/empty-state-interface-design/
- https://app.uxcel.com/courses/common-patterns/empty-states-best-practices-330
- https://uxdesign.cc/7-types-of-empty-states-and-how-to-use-them-717006dfaa8a

**Comment by halocline** [comment]
Related Slack thread: https://hpe.slack.com/archives/C04LMJWQT/p1679077250602359

Deadline: Needed early April

**Comment by jeffstolz** [comment]
I reached out to the following teams for use cases they want to be considered in the guidance. Not going to wait on them, but want to give them an opportunity to participate:

- Edgar and the entire GLCS team (sent assets)
- Derek for GLCP (sent assets)
- Christopher Kennedy-Nunez (Sent assets)
- Fred Mouveaux for Aruba
- Aideen Lawlor & Ivan O'Grady for DCE Digital Support Experience

**Comment by jeffstolz** [comment]
In progress work https://www.figma.com/file/4uEMfLl9riZDJ4y6dlv1Jq/HPE-Empty-State?node-id=1%3A280&t=hxzKNB9n3ZcoogHt-1

**Comment by jeffstolz** [comment]
4/7 - 🟢 on track. Almost ready for review.

Next steps:

- Get review/approval from team.
- Publish changes to library.
- Communication out to the org.
- Create follow up ticket for website update.

**Comment by jeffstolz** [comment]
Complete Figma file ready for review https://www.figma.com/file/4uEMfLl9riZDJ4y6dlv1Jq/HPE-Empty-State?node-id=2729%3A8940&t=zgrfUF3ru1uizJFG-1

**Comment by halocline** [comment]
@jeffstolz - I went through the draft guidance in Figma. I thought the content touched on the right elements and broke down nicely. We can refine and revise when we implement on the Design System site.

I dropped a few comments in the Figma file. Most can be addressed when implementing on the site.

From my perspective, once Kenny and Vicky have reviewed and 👍 the component implementation and an issue for implementing on the site has been created. This one can be considered 'done.'
